taca/server_status/server_status.py

```diff
@@ -158,17 +158,47 @@ def get_uppmax_cpu_hours():
     for proj in output.strip().split('\n'):
         project_dict = {"time": current_time.isoformat()}
 
-        # split line into a list
-        project = proj.split()
+        try: # split line into a list
+            project_id, cpu_hours, cpu_limit = proj.split()
         # sometimes it returns empty strings or something strange
-        try:
-            # creating objects
-            project_dict["project"] = project[0]
-            project_dict["cpu hours"] = project[1]
-            project_dict["cpu limit"] = project[2]
-            result[project[0]] = project_dict
         except Exception, e:
             logging.error(e.message)
-            continue
+        else:
+            project_dict["project"] = project_id
+            project_dict["cpu hours"] = cpu_hours
+            project_dict["cpu limit"] = cpu_limit
+
+            quotas = _get_uppmax_cpu_quotas(project_id)
+            if quotas:
+                project_dict["cpu_quota_decrease"] = quotas
+
+            result[project_id] = project_dict
+
+    return result
 
+def _get_uppmax_cpu_quotas(project_id):
+    today = datetime.date.today()
+    result = {}
+    try:
+        logging.info("CPU Quotas Decrease, project {}:".format(project_id))
+        cpu_quotas = subprocess.Popen("grep -A 10 {} /sw/uppmax/etc/projects".format(project_id).split(), stdout=subprocess.PIPE)
+        output = cpu_quotas.communicate()[0].split('\n')
+    except Exception, e:
+        logging.error(e.message)
+    else:
+        for line in output:
+            # line should be: "Shortgrants:  milou=7000	20150421"
+            if 'Shortgrants' in line:
+                logging.info(line)
+                try:
+                    shortgrants, cpu_hours, date_string = line.split()
+                except Exception, e:
+                    logging.error(e.message)
+                else:
+                    quota_date = datetime.datetime.strptime(date_string, "%Y%m%d").date()
+                    if quota_date > today:
+                        # cpu_hours will be 'milou=20000' or 'irma=20000'
+                        cpu_hours = cpu_hours.split('=')[-1]
+                        quota_date = '{}-{}-{}'.format(quota_date.year, quota_date.month, quota_date.day)
+                        result[quota_date] = cpu_hours
```

Comment:
It may be possible to have multiple grants in place, so we should have a `break` here (if one ends in 2 days and another in 2 years, this will only report the one in 2 years). This is assuming that they are listed in date order, but I think that's probably a fair assumption (until it breaks).